[mp3lame] build failure [custom triplet - universal binary]

**Host Environment**
- OS: macOS
- Compiler: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install mp3lame --triplet=x64-arm64-osx-release --overlay-triplets=/path/to/triplets`

`x64-arm64-osx-release` triplet:
<details>

```
set(VCPKG_TARGET_ARCHITECTURE x86_64 arm64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)

set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
set(VCPKG_OSX_ARCHITECTURES x86_64 arm64)
set(VCPKG_BUILD_TYPE release)
```

</details>

**Failure logs**

Output from `./vcpkg/buildtrees/mp3lame/config-x64-arm64-osx-release-supercollider-rel-err.log`:
```
Invalid configuration `universal-apple-darwin': machine `universal-apple' not recognized
configure: error: /bin/sh ./../src/0844b9eadb-a14cfeb202.clean/config.sub universal-apple-darwin failed
```



Full output of the build command:
<details>

```
Computing installation plan...
The following packages will be built and installed:
    mp3lame[core]:x64-arm64-osx-release -> 3.100#8
Detecting compiler hash for triplet x64-arm64-osx-release...
Restored 0 package(s) from /Users/marcin/.cache/vcpkg/archives in 6.291 us. Use --debug to see more details.
Installing 1/1 mp3lame:x64-arm64-osx-release...
Building mp3lame[core]:x64-arm64-osx-release...
-- [OVERLAY] Loading triplet configuration from: vcpkg/triplets/x64-arm64-osx-release.cmake
-- Using cached lame-3.100.tar.gz.
-- Cleaning sources at src/vcpkg/buildtrees/mp3lame/src/0844b9eadb-a14cfeb202.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source src/vcpkg/downloads/lame-3.100.tar.gz
-- Applying patch 00001-msvc-upgrade-solution-up-to-vc11.patch
-- Applying patch remove_lame_init_old_from_symbol_list.patch
-- Using source at src/vcpkg/buildtrees/mp3lame/src/0844b9eadb-a14cfeb202.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-arm64-osx-release-rel
-- Configuring x64-arm64-osx-release-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: /bin/bash -c "V=1 ./../src/0844b9eadb-a14cfeb202.clean/configure --host=universal-apple-darwin \"--enable-static=yes\" \"--enable-shared=no--with-pic=yes\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--prefix=vcpkg/installed/x64-arm64-osx-release\" \"--bindir=\\${prefix}/tools/mp3lame/bin\" \"--sbindir=\\${prefix}/tools/mp3lame/sbin\" \"--libdir=\\${prefix}/lib\" \"--mandir=\\${prefix}/share/mp3lame\" \"--docdir=\\${prefix}/share/mp3lame\" \"--datarootdir=\\${prefix}/share/mp3lame\""
    Working Directory: vcpkg/buildtrees/mp3lame/x64-arm64-osx-release-rel
    Error code: 1
    See logs for more information:
     vcpkg/buildtrees/mp3lame/config-x64-arm64-osx-release-rel-config.log
     vcpkg/buildtrees/mp3lame/config-x64-arm64-osx-release-rel-out.log
     vcpkg/buildtrees/mp3lame/config-x64-arm64-osx-release-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:797 (vcpkg_execute_required_process)
  ports/mp3lame/portfile.cmake:103 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)


error: building mp3lame:x64-arm64-osx-release failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+mp3lame
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[mp3lame]+Build+error
Include '[mp3lame] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-09-01-dfb82802c8cc562ce3b665a904a65b22314de724
    vcpkg-scripts version: 8e3595fad 2022-09-09 (2 days ago)

You can also use the prefilled template from vcpkg/installed/vcpkg/issue_body.md.
```
</details>

**Additional context**
`mp3lame` builds fine for single architecture `x86_64` and `arm64`. Only building for both architectures at the same time fails.

I'm guessing that mp3lame's configure script does not provide support for x64/arm64 universal binaries, but I'm not sure how that could be solved.

--- a/ports/mp3lame/add-macos-universal-config.patch
+++ b/ports/mp3lame/add-macos-universal-config.patch
@@ -1,0 +1,12 @@
+diff --git a/config.sub b/config.sub
+index 3580aaf..bf099fc 100755
+--- a/config.sub
++++ b/config.sub
+@@ -439,6 +439,7 @@ case $basic_machine in
+        | tile*-* \
+        | tron-* \
+        | ubicom32-* \
++       | universal-* \
+        | v850-* | v850e-* | v850e1-* | v850es-* | v850e2-* | v850e2v3-* \
+        | vax-* \
+        | visium-* \

--- a/ports/mp3lame/portfile.cmake
+++ b/ports/mp3lame/portfile.cmake
@@ -9,6 +9,7 @@ vcpkg_from_sourceforge(
     PATCHES
         00001-msvc-upgrade-solution-up-to-vc11.patch
         remove_lame_init_old_from_symbol_list.patch # deprecated https://github.com/zlargon/lame/blob/master/include/lame.h#L169
+        add-macos-universal-config.patch
 )
 
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)

--- a/ports/mp3lame/vcpkg.json
+++ b/ports/mp3lame/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "mp3lame",
   "version": "3.100",
-  "port-version": 8,
+  "port-version": 9,
   "description": "LAME is a high quality MPEG Audio Layer III (MP3) encoder licensed under the LGPL.",
   "homepage": "https://sourceforge.net/projects/lame",
   "license": "LGPL-2.0-only"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4798,7 +4798,7 @@
     },
     "mp3lame": {
       "baseline": "3.100",
-      "port-version": 8
+      "port-version": 9
     },
     "mpark-variant": {
       "baseline": "1.4.0",

--- a/versions/m-/mp3lame.json
+++ b/versions/m-/mp3lame.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e24c300cc7a378ed1355ca4edba742a11d0d7a7a",
+      "version": "3.100",
+      "port-version": 9
+    },
+    {
       "git-tree": "bd7f2793ec89d5ce9c00b4b9848a80905eb7ab67",
       "version": "3.100",
       "port-version": 8

